builtin/credential/userpass/path_users.go

```diff
@@ -172,25 +208,76 @@ func (b *backend) userCreateUpdate(ctx context.Context, req *logical.Request, d
 		}
 	}
 
-	if policiesRaw, ok := d.GetOk("policies"); ok {
-		userEntry.Policies = policyutil.ParsePolicies(policiesRaw)
-	}
+	// handle upgrade cases
+	{
+		policiesRaw, ok := d.GetOk("token_policies")
+		if !ok {
+			policiesRaw, ok = d.GetOk("policies")
+			if ok {
+				userEntry.Policies = policyutil.ParsePolicies(policiesRaw)
+				userEntry.TokenPolicies = nil
+			}
+		} else {
+			_, ok = d.GetOk("policies")
```

Comment:
I think each of these blocks here and below are missing a line setting the `userEntry.* = *Raw` line. e.g. 
`userEntry.TokenPolicies = policyutil.ParsePolicies(policiesRaw)

Reply:
Or is this already handled by the ParseTokenFields call?

Reply:
Already handled. This is specifically upgrade compatibility -- this is checking if they set the non token_ versions _in addition to_ the other versions and if so it's setting the non-token_ versions to be the same as the token_ versions.